rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/action/TbCopyAttributesToEntityViewNodeTest.java

```diff
@@ -0,0 +1,287 @@
+import static org.thingsboard.server.common.data.msg.TbMsgType.ATTRIBUTES_UPDATED;
+import static org.thingsboard.server.common.data.msg.TbMsgType.INACTIVITY_EVENT;
+import static org.thingsboard.server.common.data.msg.TbMsgType.POST_ATTRIBUTES_REQUEST;
+
+@ExtendWith(MockitoExtension.class)
+public class TbCopyAttributesToEntityViewNodeTest {
+
+    private final TenantId TENANT_ID = new TenantId(UUID.fromString("9fdb1f05-dc66-4960-9263-ae195f1b4533"));
+    private final DeviceId DEVICE_ID = new DeviceId(UUID.fromString("1d453dc9-9333-476a-a51f-093cf2176e59"));
+
+    private TbCopyAttributesToEntityViewNode node;
+    private EmptyNodeConfiguration config;
+
+    @Mock
+    private TbContext ctxMock;
+    @Mock
+    private EntityViewService entityViewServiceMock;
+    @Mock
+    private RuleEngineTelemetryService telemetryServiceMock;
+
+    @BeforeEach
+    void setUp() throws TbNodeException {
+        node = new TbCopyAttributesToEntityViewNode();
+        config = new EmptyNodeConfiguration().defaultConfiguration();
+        var configuration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        node.init(ctxMock, configuration);
+    }
+
+    @Test
+    public void givenExistingAttributes_whenOnMsg_thenCopyAttributesToView() {
+        EntityViewId entityViewId = EntityViewId.fromString("a2109747-d1f4-475a-baaa-55f5d4897ad8");
+        EntityView entityView = new EntityView(entityViewId);
+        entityView.setStartTimeMs(Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+        entityView.setEndTimeMs(Instant.now().plus(1, ChronoUnit.DAYS).toEpochMilli());
+        AttributesEntityView attributes = new AttributesEntityView(List.of("attribute1"), Collections.emptyList(), Collections.emptyList());
+        entityView.setKeys(new TelemetryEntityView(Collections.emptyList(), attributes));
+
+        TbMsg msg = TbMsg.newMsg(
+                TbMsgType.POST_ATTRIBUTES_REQUEST, DEVICE_ID, new TbMsgMetaData(Map.of("scope", "CLIENT_SCOPE")),
+                "{\"attribute1\": 100, \"attribute2\": \"value2\"}");
+
+        when(ctxMock.getEntityViewService()).thenReturn(entityViewServiceMock);
+        when(ctxMock.getTenantId()).thenReturn(TENANT_ID);
+        when(entityViewServiceMock.findEntityViewsByTenantIdAndEntityIdAsync(any(), any()))
+                .thenReturn(Futures.immediateFuture(List.of(entityView)));
+        when(ctxMock.getTelemetryService()).thenReturn(telemetryServiceMock);
+        doAnswer(invocation -> {
+            FutureCallback<Void> callback = invocation.getArgument(4);
+            callback.onSuccess(null);
+            return null;
+        }).when(telemetryServiceMock).saveAndNotify(any(), any(), any(AttributeScope.class), anyList(), any(FutureCallback.class));
+        doAnswer(invocation -> {
+            TbMsg newMsg = TbMsg.newMsg(msg.getQueueName(), msg.getType(), entityViewId, msg.getCustomerId(), msg.getMetaData(), msg.getData());
+            return newMsg;
```

Comment:
1. inline variable.
2. Please use non-deprecated methods if possible. see newMsg method that uses msg.getInteralType()